plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/query/FilterToPinotSqlConverter.java

```diff
@@ -0,0 +1,188 @@
+        }
+        else if (functionCall.getOperator().equalsIgnoreCase("in") || functionCall.getOperator().equalsIgnoreCase("not_in")) {
+            return parseInClause(functionCall);
+        }
+        else if (functionCall.getOperator().equalsIgnoreCase("case")) {
+            return parseCaseStatement(functionCall);
+        }
+        return functionCall.getOperator() + "(" + functionCall.getOperands().stream().map(this::formatExpression).collect(joining(", ")) + ")";
+    }
+
+    // Pinot parses "a = b" as "a - b = 0" which can result in invalid sql
+    private Optional<String> formatEqualsMinusZero(Function functionCall)
+    {
+        if (!functionCall.getOperator().equalsIgnoreCase("equals")) {
+            return Optional.empty();
+        }
+
+        Expression left = functionCall.getOperands().get(0);
+        if (left.getType() != FUNCTION || !left.getFunctionCall().getOperator().equalsIgnoreCase("minus")) {
+            return Optional.empty();
+        }
+
+        Expression right = functionCall.getOperands().get(1);
+        if (right.getType() != LITERAL || !formatLiteral(right.getLiteral()).equals("0")) {
+            return Optional.empty();
+        }
+        Function minus = left.getFunctionCall();
+        return Optional.of(format("(%s) = (%s)", formatExpression(minus.getOperands().get(0)), formatExpression(minus.getOperands().get(1))));
+    }
+
+    private String parseInClause(Function functionCall)
+    {
+        checkState(functionCall.getOperator().equalsIgnoreCase("in") ||
+                functionCall.getOperator().equalsIgnoreCase("not_in"),
+                "Unexpected operator '%s'", functionCall.getOperator());
+        checkState(functionCall.getOperands().size() > 1, "Unexpected expression");
+        String operator;
+        if (functionCall.getOperator().equalsIgnoreCase("in")) {
+            operator = "IN";
+        }
+        else {
+            operator = "NOT IN";
+        }
+        return format("%s %s (%s)", formatExpression(functionCall.getOperands().get(0)),
+                operator,
+                functionCall.getOperands().subList(1, functionCall.getOperands().size()).stream()
+                        .map(this::formatExpression)
+                        .collect(joining(", ")));
+    }
+
+    private String parseCaseStatement(Function functionCall)
+    {
+        checkState(functionCall.getOperator().equalsIgnoreCase("case"), "Unexpected operator '%s'", functionCall.getOperator());
+        checkState(functionCall.getOperands().size() >= 2, "Unexpected expression");
+        int whenStatements = functionCall.getOperands().size() / 2;
+        String caseStatement = format("CASE %s", IntStream.range(0, whenStatements)
+                .mapToObj(index -> "WHEN " + formatExpression(functionCall.getOperands().get(index)) +
+                        " THEN " + formatExpression(functionCall.getOperands().get(index + whenStatements)))
+                .collect(joining(" ")));
```

Comment:
This is a little hard to read. It might be clearer to do it with a StringBuilder and a regular for loop

Reply:
Updated